crates/bevy_ecs/src/query/state.rs

```diff
@@ -778,6 +811,157 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
             );
         }
     }
+    /// This is a batched version of `for_each_mut` that accepts a batch size, `N`, and a desired alignment for the batches `ALIGN`.
+    /// Runs `f` and `f_batch` on each query result for the given [`World`].  This can be used to improve performance of your queries using SIMD.
```

Comment:
These docs need to explain:
- what is alignment
- what is SIMD
- when this might improve (or worsen) performance

Reply:
Would it be okay if I just wrote a short blurb about these things, referring to e.g. Wikipedia or maybe some Intel blog posts for a more in-depth treatment?

Reply:
Yep, that's totally fine. I just want this to be a bit less impenetrable to users who are new to low-level programming :)

Reply:
Hey @alice-i-cecile , I added a short little introduction to `Query::for_each_mut_batched` covering these concepts, would you mind giving it a read when you get a chance?  I don't want to go overboard with info but discoverability is a good thing.  Thank you!